app/src/main/java/org/schabi/newpipe/util/PicassoHelper.java

```diff
@@ -133,28 +134,21 @@ public Bitmap transform(final Bitmap source) {
                                         .getDimension(R.dimen.player_notification_thumbnail_width),
                                 source.getWidth());
 
-                        final Bitmap result = Bitmap.createScaledBitmap(
+                        Bitmap result = BitmapCompat.createScaledBitmap(
                                 source,
                                 (int) notificationThumbnailWidth,
                                 (int) (source.getHeight()
                                         / (source.getWidth() / notificationThumbnailWidth)),
+                                null,
                                 true);
 
-                        if (result == source) {
-                            // create a new mutable bitmap to prevent strange crashes on some
-                            // devices (see #4638)
-                            final Bitmap copied = Bitmap.createScaledBitmap(
-                                    source,
-                                    (int) notificationThumbnailWidth - 1,
-                                    (int) (source.getHeight() / (source.getWidth()
-                                            / (notificationThumbnailWidth - 1))),
-                                    true);
-                            source.recycle();
-                            return copied;
-                        } else {
-                            source.recycle();
-                            return result;
+                        // create a new mutable bitmap to prevent strange crashes on some devices
+                        // (see #4638)
+                        if (!result.isMutable()) {
+                            result = result.copy(result.getConfig(), true);
```

Comment:
Nope, you can't just copy. We found out this would lead to crashes anyway (yeah, it doesn't make sense). Instead a scaled bitmap needs to be created. But anyway, let's wait for what users have to say.

Reply:
Possibly because the bitmap was a hardware bitmap, and those are always immutable ([the copy method](https://developer.android.com/reference/android/graphics/Bitmap#copy(android.graphics.Bitmap.Config,%20boolean)) throws an exception if a mutable copy is requested for such a bitmap with the hardware config). `BitmapCompat.createScaledBitmap()` creates a copy of the bitmap with a different config if it's a hardware bitmap.

Reply:
No, the copy worked correctly as far as I remember. But for some users the system crashed when that copied bitmap was displayed as the lock screen thumbnail.

Reply:
> No, the copy worked correctly as far as I remember. But for some users the system crashed when that copied bitmap was displayed as the lock screen thumbnail.

Interesting.

Maybe this line is the cause of the issue? https://github.com/androidx/androidx/blob/a66defb2607fc20c7d42ea455a0883d80033529c/palette/palette/src/main/java/androidx/palette/graphics/Palette.java#L610

Reply:
Mmmh, I think the problem was deeper than that. I think it was a crash in some C code, and since it happened only rarely on some phones, I think it was a bug in their operating systems themselves.

arielshuman said his phone has no problem, but actually, I think it's better to keep the old code and just upgrade from `Bitmap` to `BitmapCompat.createScaledBitmap` in both places where it is used. The only difference I would make is to create a new thumbnail both if `result == source` and if `!result.isMutable()` (remember to cleanup the intermediate `result` also if it is different from `source`, if the `if` is entered). Sorry for not providing a definite answer earlier, but I think it's better to be on the safe side, considering this was a bug that costed us many hours of debugging a while back.